src/main/java/com/teamrocket/service/CourierService.java

```diff
@@ -33,23 +35,25 @@ public class CourierService implements ICourierService {
     private KafkaTemplate kafkaTemplate;
 
     @Override
+    @Transactional
     public CourierDTO registerCourier(RegisterCourierRequest request) throws ResourceException {
         LOGGER.info("New register courier request {}", request);
-
+        Courier courier = new Courier(request);
         try {
-            Courier courier = new Courier(request);
-            courier = courierRepository.save(courier);
-            int userId = authClient.registerCourierUser(courier.getEmail(),courier.getId(),request.getPassword());
-            courier.setUserId(userId);
             courier = courierRepository.save(courier);
-            CourierDTO registeredCourier = new CourierDTO(courier);
-            kafkaTemplate.send(Topic.NEW_COURIER.toString(), registeredCourier);
-            return registeredCourier;
+
         } catch (DataIntegrityViolationException e) {
             LOGGER.error(e.getMessage());
             throw new ResourceException("Courier could not be saved due to an incorrect data");
         }
+        int userId = authClient.registerCourierUser(courier.getEmail(), courier.getId(), request.getPassword());
+        LOGGER.info("Registered courier with userId {}", userId);
```

Comment:
Technically I think this should be happening on the auth service side instead. However since there are no logs there then it's fine i guess.

Reply:
That's correct. I think I added that when we were trying to get auth service to work